### DISCUSSION TITLE
HardDepthShader does not support batch rendering?

### DISCUSSION THREAD

**Discussion by GritBear** [facebookresearch/pytorch3d#1374] [Q&A]
The batch rendering API which works well for other shaders, doesn't work for HardDepthShader when there are more than 1 camera angle.

Is this the expected behavior? What is the best approach to efficiently batch render a depth image?

Example exception:

RuntimeError: shape mismatch: value tensor of shape [9] cannot be broadcast to indexing result of shape [519630]

where 9 is the batch size.

**Comment by bottler** [comment]
Can you share some code which is failing? This should work.

**Comment by blacksino** [comment]
this is a still a bug in 2024
